### PR TITLE
chore(FX-3414): Update tests for pills

### PR DESCRIPTION
**Pull request by anastasiapyzhik**
The type of this PR is: **Chore**

This PR resolves [FX-3414]

### PR Checklist (tick all before merging)

- [x] I have included screenshots or videos to illustrate my changes, or I have not changed anything that impacts the UI.
- [x] I have tested my changes on **iOS** and **Android**.
- [x] I have added tests for my changes, or my changes don't require testing, or I have included a link to a separate Jira ticket covering the tests.
- [x] I have added a feature flag, or my changes don't require a feature flag. ([How do I add one?](https://github.com/artsy/eigen/blob/master/docs/developing_a_feature.md))
- [x] I have documented any follow-up work that this PR will require, or it does not require any.
- [x] I have added an app state migration, or my changes do not require one. ([What are migrations?](https://github.com/artsy/eigen/blob/master/docs/adding_state_migrations.md))
- [x] I have added a changelog entry below or my changes do not require one.

<details><summary>Changelog updates</summary>

#### Cross-platform user-facing changes

-

#### iOS user-facing changes

-

#### Android user-facing changes

-

#### Dev changes

- Update tests for search pills - anastasiapyzhik

</details>

[FX-3414]: https://artsyproduct.atlassian.net/browse/FX-3414

### CODE REVIEW

**Comment by ArtsyOpenSource** [comment]
### This PR contains the following changes:

- Cross-platform user-facing changes (Fixed pill selection <> search CTA relationship)
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against a1e1d1b38cd005bc56b6106f1417a9e1c2e2619f
</p>

File: src/lib/Scenes/Search2/Search2.tsx [hunk 1]
```diff
@@ -155,29 +156,33 @@ export const Search2: React.FC<Search2Props> = (props) => {
...
   }
 
   const isSelected = (pill: PillType) => {
     return selectedPill.name === pill.name
   }
 
   const handleSubmitEditing = () => {
-    if (shouldStartQuering) {
+    if (shouldStartQuering && query !== recentSearchQuery) {
       setSelectedPill(TOP_PILL)
```
Comment (dimatretyak) [inline src/lib/Scenes/Search2/Search2.tsx:186]: as I understood by the condition of the task, when we click on the "search" button, we just need to close the keyboard and stay on the selected pill

```javascript
const handleSubmitEditing = () => {
  if (shouldStartQuering) {
    Keyboard.dismiss()
  }
}
```

File: src/lib/Scenes/Search2/Search2.tsx [hunk 2]
```diff
@@ -155,29 +156,33 @@ export const Search2: React.FC<Search2Props> = (props) => {
...
   }
 
-  const shouldStartQuering = !!searchState?.query?.length && searchState?.query.length >= 2
+  const shouldStartQuering = !!query?.length && query.length >= 2
 
   const handlePillPress = (pill: PillType) => {
     setSearchState((prevState) => ({ ...prevState, page: 1 }))
     setSelectedPill(pill)
+    setRecentSearchQuery(query ?? "")
+    Keyboard.dismiss()
```
Comment (dimatretyak) [inline src/lib/Scenes/Search2/Search2.tsx:177]: Here it is necessary to call only `Keyboard.dismiss()`

File: src/lib/Scenes/Search2/Search2.tsx [hunk 3]
```diff
@@ -155,29 +156,33 @@ export const Search2: React.FC<Search2Props> = (props) => {
...
   }
```
Comment (dimatretyak) [inline src/lib/Scenes/Search2/Search2.tsx:178]: We should only hide the keyboard here

```javascript
const handlePillPress = (pill: PillType) => {
  setSearchState((prevState) => ({ ...prevState, page: 1 }))
  setSelectedPill(pill)
  Keyboard.dismiss()
}
```

**Comment by dimatretyak** [comment]
You can find an example of changes and demo video in PR #5567

**Comment by anastasiapyzhik** [comment]
> You can find an example of changes and demo video in PR #5570

Merged this one, thanks Dima! So current PR just updates tests for code changes

**Review by dimatretyak — APPROVED** [review]

**Pull request merged**